archey/entry.py

```diff
@@ -26,7 +27,18 @@ def __init__(self, name: Optional[str] = None, value=None, options: Optional[dic
         # `name`: key (defaults to the instantiated entry class name);
         # `value`: value of entry as an appropriate object;
         # `options`: configuration options *specific* to an entry instance;
-        self.name = name or self._PRETTY_NAME or self.__class__.__name__
+
+        configuration = Configuration()
+        icon = configuration.get("entries_icon")
+
+        if icon == True:
+            if self._ICON == None:
+                self.name = name or self._PRETTY_NAME or self.__class__.__name__
+            else:
+                self.name = name or self._PRETTY_NAME or self.__class__.__name__
+                self.name = self._ICON + " " + self.name
+        else:
+            self.name = name or self._PRETTY_NAME or self.__class__.__name__
```

Comment:
```suggestion

        if self._ICON is not None and Configuration().get("entries_icon"):
            self.name = f"{self._ICON} {self.name}"

```

Such a block could be potentially set below attribute assignments and `configuration = Configuration()` could be added for DRY purpose.

Reply:
@HorlogeSkynet this suggestion also needs to re-add `self.name = name or self._PRETTY_NAME or self.__class__.__name__` above :smile: